drivers/platform/maxim/maxim_spi.c

```diff
@@ -77,51 +78,67 @@ int32_t max_spi_init(struct no_os_spi_desc **desc,
 		return -ENOMEM;
 
 	eparam = param->extra;
+
 	descriptor->device_id = param->device_id;
 	descriptor->max_speed_hz = param->max_speed_hz;
 	descriptor->chip_select = param->chip_select;
 	descriptor->mode = param->mode;
 	descriptor->bit_order = param->bit_order;
 	descriptor->platform_ops = &max_spi_ops;
 
-	if (descriptor->device_id == 0)
-		ret = MXC_SPI_Init(MXC_SPI0, SPI_MASTER_MODE, SPI_SINGLE_MODE,
-				   eparam->numSlaves, eparam->polarity, param->max_speed_hz);
-#ifdef MXC_SPI1
-	else if (descriptor->device_id == 1)
-		ret = MXC_SPI_Init(MXC_SPI1, SPI_MASTER_MODE, SPI_SINGLE_MODE,
-				   eparam->numSlaves, eparam->polarity, param->max_speed_hz);
-#endif
-	else {
+	if (descriptor->device_id >= MXC_SPI_INSTANCES) {
 		ret = -EINVAL;
 		goto err;
 	}
+		
+#if TARGET_NUM == 32655
```

Comment:
what is `32655`? A relevant macro might help users/reviewers understand.

Reply:
32655 is the target that the user builds for (in this case MAX32655). Would MAX32655 or something similar be more clear?

Reply:
Got it. I guess it's your call if you want or not to add macros.